plasmapy/utils/decorators/converter.py

```diff
@@ -0,0 +1,46 @@
+"""
+Decorator to convert units of functions in /physics methods
+"""
+__all__ = ["angular_freq_to_hz"]
+
+import functools
+import inspect
+from astropy import units as u
+from plasmapy.utils.decorators import preserve_signature
+
+
+def angular_freq_to_hz(fn):
+    # raise exception if fn uses the 'to_hz' kwarg
+    sig = inspect.signature(fn)
+    if 'to_hz' in sig.parameters:
+        raise raise ValueError(f"Wrapped function '{fn.__name__}'
+        can not use keyword 'to_hz'.  Keyword reserved for decorator functionality.")
+
+    # make new signature for fn
+    new_params = sig.parameters.copy()
+    new_params['to_hz'] = inspect.Parameter('to_hz', inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                                            default=False)
+    new_sig = inspect.Signature(parameters=new_params.values(),
+                                return_annotation=sig.return_annotation)
+    fn.__signature__ = new_sig
+
+    # add 'to_hz' to fn docstring
+    fn.__doc__ = """
+    Other Parameters
+    ----------------
+    to_hz: bool
+        Set `True` to to convert function output from angular frequency to Hz
+    """
+    fn.__doc__ += """
```

Comment:
you've added to `fn.__doc__` twice...delete one instance...note for section heading that the "-" line must match the length of the heading, that is...
```
Other Parameters
----------------
```

not 

```
Other Parameters
-------------
```
nor
```
Other Parameters
---------------------
```

Reply:
thank you for that, im pushing code again

Reply:
so, the idea is to put the to_hz = True/False as an argument on the decorator?. Because the line
`if 'to_hz' in sig.parameters:`
will crash in the current way 
`def lower_hybrid_frequency(B, n_i, ion='p+', to_hz=False):`

Reply:
This is the point.  We're enforcing that the `to_hz` keyword is reserved for the decorator and, thus, the function being wrapped should NOT use it.  None of the function definitions should declare the `to_hz` keyword.

Reply:
So how are we going to do the conversion? if we use it inside another function like `lower_hybrid_frequency` which is using this 3 functions 
`    omega_ci = gyrofrequency(B, particle=ion)
    omega_pi = plasma_frequency(n_i, particle=ion)
    omega_ce = gyrofrequency(B)`
It should return with rad / s, but if we called from 
`gyrofrequency(0.1*u.T)` 
in this case should it return in hz by itself?

Reply:
I'm not sure what issue you're trying to point out.  The definition for all characteristic plasma frequencies (lower-hybrid, plasma frequency, gyro-frequencies, etc.) are in terms of `rads/s`.  Thus, any defined "frequency" function should, by default, return a quantity in units of `rads/s`.  The decorator just takes that result and converts it to `Hz` when requested.  So...

`gyrofrequency(0.1*u.T)` will be in `rads/s`
`gyrofrequency(0.1*u.T, to_hz=True)` will be in `Hz`

In the `to_hz` instance, the calculation inside the `gyrofrequency()` function is still done in `rads/s` and then converted by the decorator.  The function never even sees the `to_hz` keyword.